tachyon/crypto/commitments/fri/two_adic_fri.h

```diff
@@ -265,19 +265,19 @@ class TwoAdicFRI {
                   index >> bits_reduced, log_num_rows);
               F w;
               CHECK(F::GetRootOfUnity(size_t{1} << log_num_rows, &w));
-              F x = F::FromMontgomery(F::Config::kSubgroupGenerator) *
-                    w.Pow(rev_reduced_index);
+              ExtF x(F::FromMontgomery(F::Config::kSubgroupGenerator) *
+                     w.Pow(rev_reduced_index));
 
               auto it = reduced_openings.try_emplace(
                   log_num_rows, std::make_tuple(ExtF::One(), ExtF::Zero()));
               std::tuple<ExtF, ExtF>& reduced_opening = it.first->second;
               for (size_t i = 0; i < cur_points.size(); ++i) {
                 const ExtF& z = cur_points[i];
+                ExtF denom = unwrap((x - z).Inverse());
                 const std::vector<ExtF>& ps_at_z = cur_values[i];
                 CHECK_EQ(ps_at_z.size(), cur_values_in.size());
                 for (size_t j = 0; j < ps_at_z.size(); ++j) {
-                  ExtF quotient = unwrap((ExtF(cur_values_in[j]) - ps_at_z[j]) /
-                                         (ExtF(x) - z));
+                  ExtF quotient = (ExtF(cur_values_in[j]) - ps_at_z[j]) * denom;
```

Comment:
What about also minimizing conversions from f to extf by initializing cur_values_in to extfs